python/tvm/relay/qnn/op/qnn.py

```diff
@@ -72,3 +73,75 @@ def requantize(data,
                             output_zero_point,
                             rounding,
                             out_dtype)
+
+def concatenate(data,
+                input_scales,
+                input_zero_points,
+                output_scale,
+                output_zero_point,
+                axis):
+    """Concatenate the quantized input tensors along the given axis.
+
+    Parameters
+    ----------
+    data : Union(List[relay.Expr], Tuple[relay.Expr])
+        The list of quantized tensors.
+
+    input_scales : List[float32]
+        The list of scales of input quantized tensors.
+
+    input_zero_points : List[int32]
+        The list of zero points of input quantized tensors.
+
+    output_scale : float32
+        The scale of the output quantized tensor.
+
+    output_zero_point : int32
+        The zero point of the output quantized tensor.
+
+    axis : int
+        The axis along which the tensors are concatenated.
+
+    Returns
+    -------
+    result: relay.Expr
+        The concatenated quantized tensor.
+    """
+
+    data = list(data)
+    requantized_exprs = list(data)
+
+    # Find the dtype of the input expr. This is required for the requantize op. Since, this is
+    # concatenate op, the dtype of the input is same as dtype of the output.
+    data0 = relay.transform.infer_type(data[0])
```

Comment:
@anijain2305 This API has been removed in favor of `InferType`, please update and send another PR

Reply:
Just added a PR for fixing InferType. I think it missed CI because CI passed 3 days ago and InferType changes happened after that.

Regarding `requantize`, it makes sense. Let me think little more tomorrow and finalize this.

Reply:
I am trying to understand if we need to do type inference here. Shouldn't we have type for the input expr already? If not, shouldn't we infer type first?